common/src/redis/redis.service.ts

```diff
@@ -46,20 +47,24 @@ export class RedisService {
         await this.redis.send_command("json.set", key, path, JSON.stringify(input));
     }
 
-    async getJson<T>(key: string, path?: string): Promise<T> {
+    async getJson<T, S extends ZodSchema = ZodSchema>(key: string, path?: string, schema?: S): Promise<T> {
         const args: string[] = [key];
         if (path) {
             args.push(path);
         }
-        return JSON.parse(await this.redis.send_command("json.get", ...args) as string) as T;
+
+        const data = JSON.parse(await this.redis.send_command("json.get", ...args) as string) as unknown;
+
+        return (schema ? schema.parse(data) : data) as T;
+        
     }
 
-    async getJsonIfExists<T>(key: string): Promise<T | undefined> {
+    async getJsonIfExists<T, S extends ZodSchema = ZodSchema>(key: string, schema?: S): Promise<T | null> {
         try {
-            const obj = JSON.parse(await this.redis.send_command("json.get", key) as string) as T;
-            return obj;
+            const data = JSON.parse(await this.redis.send_command("json.get", key) as string) as unknown | null;
+            return data ? (schema ? schema.parse(data) : data) as T : null;
```

Comment:
Can we clean this up?

Nested ternaries gets a little hard to read all on one line.